### ISSUE TITLE
[quant] PBO/CSCV statistical-power floor at small library N

### ISSUE THREAD

**Issue by onder-akkaya** [aprin-labs/archimedes#819]
## Summary

`compute_pbo` (`_rigor_helpers.py`) is a faithful Combinatorially-Symmetric Cross-Validation
(CSCV) estimator of the Probability of Backtest Overfitting (Bailey, Borwein, López de Prado &
Zhu 2014): `C(16,8)=12,870` IS/OOS splits, IS-optimal config ranked OOS, logit of the relative
rank, `PBO = #{λ ≤ 0}/#splits`. The mechanics are correct.

The concern is **statistical power at our operating point**, not correctness. CSCV was designed
to assess a *single strategy's parameter sweep*, where the cross-section is hundreds of
configuration variants. We apply it across the **heterogeneous strategy library**, whose size
is currently N ≈ 4–6. At N = 4 the OOS relative rank `ω = rank/N` takes only ~4 distinct values,
so the logit `λ = ln(ω/(1−ω))` is coarsely quantized and `PBO` is granular and **library-coupled**:
adding or removing one neighbour can flip a member's verdict. The estimator is consistent as
N → ∞ but underpowered and unstable at single-digit N — and it currently *gates* Tier-1 admission
(criterion 4).

The docstring already flags the off-label use honestly, so this is not a claim-integrity bug;
it is a request to make the gate's behaviour at small N statistically defensible.

## Proposed resolution

Introduce a minimum-N floor for PBO **gating** (not computation):

- Below the floor (e.g. N < ~10–12; exact value to be justified from the logit granularity),
  report PBO as `NOT_RUN` with an explicit reason — mirroring the CPCV `NOT_RUN` treatment
  added in #771 — and let criterion 4 be neutral/skipped rather than gating on an underpowered
  statistic.
- At or above the floor, gate as today.
- Still **compute and surface** PBO below the floor (advisory), so the number is visible
  without being load-bearing.

The floor value is the statistics call here; I'll justify it from the quantization of the
relative-rank distribution and a small power simulation.

## Scope (exact files)

- `backend/archimedes/services/_rigor_helpers.py` (`compute_pbo` / its callers) and
  `backend/archimedes/services/rigor_evaluator.py` (`RigorGateResult.passes_all` /
  `gate_details["pbo"]`) — add the N-floor gating logic + honest `NOT_RUN` surface.

## Acceptance criteria

- [ ] With a library below the floor, `gate_details["pbo"]` reads `NOT_RUN (N=… below power floor)`
      and criterion 4 does not gate; tests pin this.
- [ ] At/above the floor, behaviour is unchanged; existing PBO tests still pass.
- [ ] A short power note (docstring or `docs/specs/selection-bias-corrections-spec.md`) records
      why the floor sits where it does.

## Verify

```bash
pytest backend/tests/test_rigor_evaluator.py backend/tests/test_selection_bias_routes.py -q
```

## Anti-goals

- Do not change the CSCV math itself — it is correct; only its gating applicability at small N.
- Do not stop *computing* PBO below the floor — keep it advisory and visible.

## References

- Bailey, D., Borwein, J., López de Prado, M. & Zhu, Q. (2014). *The Probability of Backtest
  Overfitting.* Journal of Computational Finance.
- Casella, G. & Berger, R. (2002). *Statistical Inference* (2nd ed.) — finite-sample vs
  asymptotic behaviour of rank-based statistics.
